here-naksha-app-service/src/test/java/com/here/naksha/app/service/DeleteFeatureTestHelper.java

```diff
@@ -0,0 +1,131 @@
+import com.here.naksha.lib.core.models.naksha.Space;
+import java.net.http.HttpResponse;
+import java.util.UUID;
+import org.jetbrains.annotations.NotNull;
+import org.skyscreamer.jsonassert.JSONAssert;
+import org.skyscreamer.jsonassert.JSONCompareMode;
+
+public class DeleteFeatureTestHelper {
+
+  final @NotNull NakshaApp app;
+  final @NotNull NakshaTestWebClient nakshaClient;
+
+  public DeleteFeatureTestHelper(final @NotNull NakshaApp app, final @NotNull NakshaTestWebClient nakshaClient) {
+    this.app = app;
+    this.nakshaClient = nakshaClient;
+  }
+
+  void tc0900_testDeleteFeatures() throws Exception {
+    // Test API : DELETE /hub/spaces/{spaceId}/features
+    final String streamId = UUID.randomUUID().toString();
+
+    // Preparation: create storage, event handler, space, and initial features
+    final String storage = loadFileOrFail("TC0900_deleteFeatures/create_storage.json");
+    nakshaClient.post("hub/storages", storage, streamId);
+    final String handler = loadFileOrFail("TC0900_deleteFeatures/create_handler.json");
+    nakshaClient.post("hub/handlers", handler, streamId);
+    final String spaceJson = loadFileOrFail("TC0900_deleteFeatures/create_space.json");
+    nakshaClient.post("hub/spaces", spaceJson, streamId);
+    final Space space = parseJsonFileOrFail("TC0900_deleteFeatures/create_space.json", Space.class);
+    final String createFeaturesJson = loadFileOrFail("TC0900_deleteFeatures/create_features.json");
+    nakshaClient.post("hub/spaces/" + space.getId() + "/features", createFeaturesJson, streamId);
+    final String expectedBodyPart = loadFileOrFail("TC0900_deleteFeatures/response.json");
+
+    // When: request is submitted to NakshaHub Space Storage instance
+    final HttpResponse<String> response = nakshaClient.delete(
+        "hub/spaces/" + space.getId() + "/features?id=feature-1-to-delete&id=feature-2-to-delete", streamId);
+
+    // Then: Perform assertions
+    assertEquals(200, response.statusCode(), "ResCode mismatch");
+    JSONAssert.assertEquals(
+        "Delete Feature response body doesn't match",
+        expectedBodyPart,
+        response.body(),
+        JSONCompareMode.LENIENT);
+    assertEquals(streamId, getHeader(response, HDR_STREAM_ID), "StreamId mismatch");
+  }
+
+  void tc0901_testDeleteNonExistingFeatures() throws Exception {
+    // Test API : DELETE /hub/spaces/{spaceId}/features
+    final String streamId = UUID.randomUUID().toString();
+
+    final Space space = parseJsonFileOrFail("TC0900_deleteFeatures/create_space.json", Space.class);
+    final String expectedBodyPart = loadFileOrFail("TC0901_deleteNonExistingFeatures/response.json");
+
+    // When: request is submitted to NakshaHub Space Storage instance
+    final HttpResponse<String> response = nakshaClient.delete(
+        "hub/spaces/" + space.getId() + "/features?id=non-existing-phantom-feature", streamId);
+
+    // Then: Perform assertions
+    assertEquals(404, response.statusCode(), "ResCode mismatch");
```

Comment:
For response, which returns FeatureCollection, we should return 2xx with empty `features` and empty `deleted` array, when feature to be deleted is not found.
For response which returns single Feature, we can return 404, when feature to be deleted is not found.

Reply:
Ok, I have adjusted the code. For the empty collection response, only the empty `features` array is returned and not the `deleted` array, as agreed through direct message. This is also aligned with other APIs.